packages/mobile-native-mcp-server/src/tools/magi/prd/magi-prd-orchestrator/tool.ts

```diff
@@ -0,0 +1,308 @@
+        taskId: interruptedTask.id,
+        interrupts: interruptedTask.interrupts.length,
+        userInput: input.userInput,
+      });
+
+      // Resume workflow with user input from previous tool execution
+      result = await compiledWorkflow.invoke(
+        new Command({ resume: input.userInput }),
+        threadConfig
+      );
+
+      this.logger.info('Resumed workflow result', { result });
+    } else {
+      // Start new PRD workflow session
+      this.logger.info('Starting new PRD workflow execution');
+
+      // Initialize state with user input - validation will be handled by workflow nodes
+      const initialState: Partial<PRDState> = {
+        userInput: input.userInput,
+        functionalRequirements: [],
+        gapAnalysisScore: 0,
+        identifiedGaps: [],
+        shouldIterate: false,
+        userIterationOverride: undefined,
+        prdContent: '',
+        prdStatus: {
+          author: 'PRD Generator',
+          lastModified: new Date().toISOString(),
+          status: 'draft' as const,
+        },
+        isPrdApproved: false,
+      };
+
+      result = await compiledWorkflow.invoke(initialState, threadConfig);
+    }
+
+    this.logger.debug('Processing PRD workflow result');
+    graphState = await compiledWorkflow.getState(threadConfig);
+    if (graphState.next.length > 0) {
+      // There are more nodes to execute.
+      const mcpToolInvocationData: MCPToolInvocationData<z.ZodObject<z.ZodRawShape>> | undefined =
+        '__interrupt__' in result
+          ? (
+              result.__interrupt__ as Array<{
+                value: MCPToolInvocationData<z.ZodObject<z.ZodRawShape>>;
+              }>
+            )[0].value
+          : undefined;
+
+      if (!mcpToolInvocationData) {
+        this.logger.error('PRD workflow completed without expected MCP tool invocation.');
+        throw new Error('FATAL: Unexpected PRD workflow state without an interrupt');
+      }
+
+      if (mcpToolInvocationData) {
+        this.logger.info('PRD workflow interrupted for tool execution', {
+          toolName: mcpToolInvocationData.llmMetadata.name,
+        });
+
+        // Save workflow state for resumption
+        await this.saveCheckpointerState(checkpointer, threadId);
+
+        // Create orchestration prompt
+        const orchestrationPrompt = this.createOrchestrationPrompt(mcpToolInvocationData, {
+          thread_id: threadId,
+        });
+
+        return {
+          orchestrationInstructionsPrompt: orchestrationPrompt,
+        };
+      }
+    }
+
+    // Workflow completed or no interruption
+    this.logger.info('PRD workflow completed', {
+      threadId,
+    });
+
+    return {
+      orchestrationInstructionsPrompt:
+        'The PRD generation workflow has concluded. No further workflow actions are forthcoming.',
+    };
+  }
+
+  /**
+   * Save checkpointer state for resumption
+   */
+  private async saveCheckpointerState(
```

Comment:
There are some differences in these two methods, between this orchestrator and the original. Are there changes here that you'd want to persist to the common orchestrator to come?

Reply:
I had some drift between the two orchestrators because my changes have a large integration test that makes multiple tool calls so I wanted state persistence between the tool calls - but maybe for integration tests we should actually use disk persistence -- something to discuss for the abstracted version....